
When altivec is disabled there is no swScale_altivec ...

Additionally cleaned up the else case a bit.

  - Rene Rebe <rene@rocklinux.org>

--- MPlayer-1.0pre4/postproc/swscale.c	2004-04-26 21:38:17.000000000 +0200
+++ MPlayer-1.0pre4-fixed/postproc/swscale.c	2004-04-28 15:02:29.000000000 +0200
@@ -1339,19 +1339,16 @@
 		return swScale_3DNow;
 	else if(flags & SWS_CPU_CAPS_MMX)
 		return swScale_MMX;
-	else
-		return swScale_C;
 
 #else
 #ifdef ARCH_POWERPC
+#ifdef HAVE_ALTIVEC
 	if(flags & SWS_CPU_CAPS_ALTIVEC)
 	  return swScale_altivec;
-	else
-	  return swScale_C;
 #endif
-	return swScale_C;
 #endif
-#else //RUNTIME_CPUDETECT
+#endif
+#else //!RUNTIME_CPUDETECT
 #ifdef HAVE_MMX2
 	return swScale_MMX2;
 #elif defined (HAVE_3DNOW)
@@ -1360,10 +1357,9 @@
 	return swScale_MMX;
 #elif defined (HAVE_ALTIVEC)
 	return swScale_altivec;
-#else
-	return swScale_C;
 #endif
 #endif //!RUNTIME_CPUDETECT
+	return swScale_C;
 }
 
 static int PlanarToNV12Wrapper(SwsContext *c, uint8_t* src[], int srcStride[], int srcSliceY,
